### ISSUE TITLE
When launching nvim: *** stack smashing detected ***:  <unknown> terminated

### ISSUE THREAD

**Issue by Thytom** [neovim/neovim#9498]
- `nvim --version`: 0.3.4-1
- Vim (version: 8.1.0570-1) behaves differently? Yes
- Operating system/version: Arch Linux (4.20.0-arch1-1-ARCH)
- Terminal name/version: URxvt 9.22-7
- `$TERM`: rxvt-unicode-256color

### Steps to reproduce using `nvim -u NORC`

```
nvim -u NORC

```

### Actual behaviour

archie@Aspire:~$ nvim -u NORC
*** stack smashing detected ***: <unknown> terminated
Aborted (core dumped)

### Expected behaviour

Launches nvim

Hello,

I believe this started happening completely randomly as nvim has been working fine all day. I did run a system update, however I don't remember if I used nvim or not after. Rolling back to v0.3.3-1 does not fix the issue. I'd be happy to provide any more info when needed. Thanks.

**Comment by justinmk** [comment]
did you update your system recently? 

Seems related to https://github.com/neovim/neovim/issues/223 , maybe distros are more aggressively enabling such things. If that's the case then this would also affect Vim.

**Comment by RatanRSur** [comment]
There's a fix out for arch users if they update again now

**Comment by JoshH100** [comment]
I've confirmed that the issue arrises from the 7a3913a libvterm commit. I suspected compiler flag changes but I don't see any changes. Downgrading libvterm to the previous git commit fixes the issue.

**Comment by Thytom** [comment]
Another update fixed the issue. Apologies for duplicate of #9497, it wasn't on the list when I checked. Thanks for quick response.

**Comment by roket1428** [comment]
This command fixed my issue `sudo pacman -U https://archive.archlinux.org/packages/l/libvterm/libvterm-0.1.git3.7a3913a-1-x86_64.pkg.tar.xz`

**Comment by SpaceboyRoss01** [comment]
I have this issue too.

**Comment by Thytom** [comment]
Built v0.4.0-138-g0d66cdc6f from source just in case, works fine.

**Comment by stevemolloy** [comment]
I experienced this after a pacman update of neovim ~12 hours ago.  Updating via pacman (just `pacman -Syu` as usual) fixed the issue.

**Comment by tleydxdy** [comment]
Still having this issue, Syu and reinstall didn't fix the issue.

**Comment by SpaceboyRoss01** [comment]
Updating fixed it for me.

**Comment by tleydxdy** [comment]
I must have been on a very slow mirror. It's fixed.

**Comment by sgtnasty** [comment]
A little late to the party, but I would like to add this happens in Fedora Beta 31. I cant run any nvim command, but here is some info.

```
Name         : neovim
Version      : 0.3.8
Release      : 2.fc31
Architecture : x86_64
Size         : 19 M
Source       : neovim-0.3.8-2.fc31.src.rpm

Vi (version: 8.1) behaves differently? Yes
Terminal: Gnome Terminal 3.34.0
Terminal name/version: xterm-256color

```

Fedora

```
Fedora release 31 (Thirty One)
Linux fedoravm 5.3.0-1.fc31.x86_64 #1 SMP Mon Sep 16 12:34:42 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by facchettos** [comment]
Same here on opensuse tumbleweed, just updated and cant launch vim anymore

**Comment by blueyed** [comment]
@facchettos 
Is this 0.3.8 or 0.4.2 then?

**Comment by blueyed** [comment]
Maybe libvterm was updated, but not neovim?

**Comment by facchettos** [comment]
> Maybe libvterm was updated, but not neovim?

Ok so i fixed the problem by using the opensuse leap libvterm, I assume that it is a problem from that library. 
I am not sure if i updated only one or both, but the problem is definitely coming from libvterm0

**Comment by Phill** [comment]
Fedora 31 - Got the problem the other day after upgrading.
I downloaded the source, compiled it, and it works fine.

**Comment by blueyed** [comment]
Yes, libvterm1 is not compatible with nvim < 0.4, and libvterm0 not with nvim >= 0.4.
